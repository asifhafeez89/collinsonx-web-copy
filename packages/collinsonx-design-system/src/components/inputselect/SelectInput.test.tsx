import renderer from 'react-test-renderer';
import SelectInput from '.';

describe('<SelectInput />', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders SelectInput', () => {
    const tree = renderer
      .create(
        <SelectInput
          label="Your favorite framework/library"
          placeholder="Pick one"
          options={[
            { value: 'react', label: 'React' },
            { value: 'ng', label: 'Angular' },
            { value: 'svelte', label: 'Svelte' },
            { value: 'vue', label: 'Vue' },
          ]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
