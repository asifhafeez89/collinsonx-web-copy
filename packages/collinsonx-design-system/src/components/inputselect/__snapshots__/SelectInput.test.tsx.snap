// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectInput /> renders SelectInput 1`] = `
.emotion-2 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  line-height: 1.55;
  color: #000!important;
}

.emotion-2 label {
  color: #000000;
  font-weight: 400;
  margin-bottom: 8px;
}

.emotion-2 input[data-invalid] {
  border-color: red;
}

.emotion-5 {
  display: inline-block;
  font-size: 0.875rem;
  font-weight: 500;
  color: #212529;
  word-break: break-word;
  cursor: default;
  -webkit-tap-highlight-color: transparent;
}

.emotion-8 {
  position: relative;
}

.emotion-11 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  height: 2.25rem;
  -webkit-tap-highlight-color: transparent;
  line-height: calc(2.25rem - 0.125rem);
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  resize: none;
  box-sizing: border-box;
  font-size: 0.875rem;
  width: 100%;
  color: #000;
  display: block;
  text-align: left;
  border: 0.0625rem solid #ced4da;
  background-color: #fff;
  -webkit-transition: border-color 100ms ease;
  transition: border-color 100ms ease;
  min-height: 2.25rem;
  padding-left: calc(2.25rem  / 3);
  padding-right: 2.25rem;
  border-radius: 0.25rem;
}

.emotion-11:focus,
.emotion-11:focus-within {
  outline: none;
  border-color: #228be6;
}

.emotion-11:disabled,
.emotion-11[data-disabled] {
  background-color: #f1f3f5;
  color: #909296;
  opacity: 0.6;
  cursor: not-allowed;
}

.emotion-11:disabled::-webkit-input-placeholder {
  color: #909296;
}

.emotion-11:disabled::-moz-placeholder {
  color: #909296;
}

.emotion-11:disabled:-ms-input-placeholder {
  color: #909296;
}

.emotion-11[data-disabled]::-webkit-input-placeholder {
  color: #909296;
}

.emotion-11[data-disabled]::-moz-placeholder {
  color: #909296;
}

.emotion-11[data-disabled]:-ms-input-placeholder {
  color: #909296;
}

.emotion-11:disabled::placeholder,
.emotion-11[data-disabled]::placeholder {
  color: #909296;
}

.emotion-11[data-invalid] {
  color: #fa5252;
  border-color: #fa5252;
}

.emotion-11[data-invalid]::-webkit-input-placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-11[data-invalid]::-moz-placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-11[data-invalid]:-ms-input-placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-11[data-invalid]::placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-11[data-with-icon] {
  padding-left: 2.25rem;
}

.emotion-11::-webkit-input-placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-11::-moz-placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-11:-ms-input-placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-11::placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-11::-webkit-inner-spin-button,
.emotion-11::-webkit-outer-spin-button,
.emotion-11::-webkit-search-decoration,
.emotion-11::-webkit-search-cancel-button,
.emotion-11::-webkit-search-results-button,
.emotion-11::-webkit-search-results-decoration {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
}

.emotion-11[type=number] {
  -moz-appearance: textfield;
}

.emotion-11:not(:disabled) {
  cursor: pointer;
}

.emotion-11:not(:disabled)::selection {
  background-color: transparent;
}

.emotion-12 {
  position: absolute;
  top: 0;
  bottom: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 2.25rem;
}

<div
  className="emotion-0 emotion-1 emotion-2"
>
  <label
    className="emotion-3 emotion-4 emotion-5"
    htmlFor="mantine-w9n6x6m84"
    id="mantine-w9n6x6m84-label"
    onMouseDown={[Function]}
  >
    Your favorite framework/library
  </label>
  <div
    aria-controls="mantine-w9n6x6m84"
    aria-expanded={false}
    aria-haspopup="listbox"
    aria-owns={null}
    onMouseLeave={[Function]}
    role="combobox"
    tabIndex={-1}
  >
    <input
      disabled={false}
      type="hidden"
      value=""
    />
    <div
      className="emotion-6 emotion-7 emotion-8"
    >
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-controls={null}
        aria-invalid={false}
        autoComplete="off"
        className="emotion-9 emotion-10 emotion-11"
        data-mantine-stop-propagation={false}
        disabled={false}
        id="mantine-w9n6x6m84"
        name={null}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        placeholder="Pick one"
        readOnly={true}
        required={false}
        type="search"
        value=""
      />
      <div
        className="emotion-12 emotion-13 emotion-14"
      >
        <test-file-stub />
      </div>
    </div>
  </div>
</div>
`;
